tests/Core/Tokenizer/EnumTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Tests the support of PHP 8.1 "enum" keyword.
+ *
+ * @author    Jaroslav Hanslík <kukulich@kukulich.cz>
+ * @copyright 2021 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Tokenizer;
+
+use PHP_CodeSniffer\Tests\Core\AbstractMethodUnitTest;
+
+class EnumTest extends AbstractMethodUnitTest
+{
+
+
+    /**
+     * Test that the "enum" keyword is tokenized as such.
+     *
+     * @param string $testMarker              The comment which prefaces the target token in the test file.
+     * @param int    $expectedScopeOpenerLine Expected line of scope opener.
+     * @param int    $expectedScopeCloserLine Expected line of scope closer.
+     *
+     * @dataProvider dataEnums
+     * @covers       PHP_CodeSniffer\Tokenizers\PHP::tokenize
+     *
+     * @return void
+     */
+    public function testEnums($testMarker, $expectedScopeOpenerLine, $expectedScopeCloserLine)
```

Comment:
Just wondering - why test the _line_ and not the actual token position (in relation to the keyword token) ?
I realize the latter is more fiddly for setting up the tests, but it's also much more precise.

In other tests, this has been done by the data provider providing an `offset` and the token position then being calculated by `$expectedOpener = $enum + $openerOffset;`.

Reply:
Modified.

Reply:
Looks like you modified it for scope opener, but removed the check for scope closer ?

Reply:
Added now.